python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -236,12 +233,21 @@ def evaluate(
             # Filter groups by min_query_count
             df_grouped_stats = df_grouped_stats[
                 df_grouped_stats["query_count"] >= group_metrics_min_queries
-            ]
+                ]
 
             # Compute group metrics
             df_group_metrics = df_grouped_stats.apply(
                 metrics_helper.summarize_grouped_stats, axis=1
             )
+            # Add power analysis to group metric dataframe
+            if len(group_key) > 0:
+                df_group_metrics = df_group_metrics.reset_index()
+                if len(group_key) > 1:
+                    df_group_metrics[str(group_key)] = df_group_metrics[group_key].apply(tuple, axis=1)
+                df_group_metrics[str(group_key)] = df_group_metrics[group_key]
+                df_group_metrics = pd.merge(df_group_metrics, group_metrics_stat_sig, on=str(group_key), how='left').drop(
+                    columns=[str(group_key)])
+
```

Comment:
@mohazahran Can you explain why we need to do this separately?

Reply:
because it's hard to modify `get_grouped_stats` to compute var/mean/count for a variable number of metrics without greatly changing the function altogether. so for now I will keep the power analysis as a parallel task